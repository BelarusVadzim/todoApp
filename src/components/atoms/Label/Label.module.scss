@import "src/variables.scss";

$regular-label-color: $secondary-font-color;
$lineThrough-label-color: $inactive-color;

.label {
  font-size: $font-size;
  color: $regular-label-color;
  border: none;
  font-family: $main-font;
  cursor: default;
  min-width: none;
}

.labelLineThrough {
  text-decoration: line-through;
  color: lighten($lineThrough-label-color, 10%);

  @extend .label;
}

