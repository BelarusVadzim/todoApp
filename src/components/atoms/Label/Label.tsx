import React from 'react';
import { PropsWithChildren } from 'react';
import style from './Label.module.scss';
import { PropsWithClassName } from 'types';

type LabelProps = PropsWithChildren &
PropsWithClassName & {
  lineThrough?: boolean;
  draggable?: boolean;
};

const Label: React.FC<LabelProps> = ({
  children,
  lineThrough,
  className = '',
  draggable,
}) => {
  let combinedClassName = lineThrough ? style.labelLineThrough : style.label;
  combinedClassName = className
    ? `${combinedClassName} ${className}`
    : combinedClassName;

  return (
    <div
      {...(draggable ? { 'data-draggable': 'true' } : {})}
      className={combinedClassName}
    >
      {children}
    </div>
  );
};

export default Label;
