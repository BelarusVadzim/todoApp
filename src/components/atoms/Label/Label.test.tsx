import Label from './Label';
import { render } from '@testing-library/react';

jest.mock('./Label.module.scss', () => ({
  labelLineThrough: 'labelLineThroughstyle',
  label: 'labelStyle',
}));

describe('<Label />', () => {
  it('should render properly', () => {
    const component = render(<Label>LabelChild</Label>);

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('when draggable lineThrough className specified should render properly', () => {
    const component = render(
      <Label draggable lineThrough className="testClass">
        LabelChild
      </Label>
    );

    expect(component.asFragment()).toMatchSnapshot();
  });
});
