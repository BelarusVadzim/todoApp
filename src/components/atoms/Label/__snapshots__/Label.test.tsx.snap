// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="labelStyle"
  >
    LabelChild
  </div>
</DocumentFragment>
`;

exports[`<Label /> when draggable lineThrough className specified should render properly 1`] = `
<DocumentFragment>
  <div
    class="labelLineThroughstyle testClass"
    data-draggable="true"
  >
    LabelChild
  </div>
</DocumentFragment>
`;
